### ISSUE TITLE
Chapter 2 comments

### ISSUE THREAD

**Issue by RaphaelS1** [mlr-org/mlr3book#439]
## Style

* [x] This could be massively shortened by combining all code in one chunk than explaining it in text
https://github.com/mlr-org/mlr3book/blob/ac3f3b95eee0f4d1556734d83d05860596ad4c16/book/basics.qmd#L533-L550
* [x] Consider `lm` over `rpart` as model output is considerably less and therefore neater
https://github.com/mlr-org/mlr3book/blob/ac3f3b95eee0f4d1556734d83d05860596ad4c16/book/basics.qmd#L527
* [x] Don't list all learners and measures just the `head`
https://github.com/mlr-org/mlr3book/blob/ac3f3b95eee0f4d1556734d83d05860596ad4c16/book/basics.qmd#L454-L461
https://github.com/mlr-org/mlr3book/blob/ac3f3b95eee0f4d1556734d83d05860596ad4c16/book/basics.qmd#L689

## Content

* [x] Because we distinguish between Tasks and Data in mlr, consider splitting them in the diagram. I don't think Repeat/Resample is needed at this stage, it's a bit confusing to look at without context (it isn't clear how this relates to using the same or different data/measure/model etc.). Also move figure after the text at the end of the section preamble.
https://github.com/mlr-org/mlr3book/blob/ac3f3b95eee0f4d1556734d83d05860596ad4c16/book/basics.qmd#L10
* [x] Need a clear discussion between difference between learner and model, currently I think a bit too brief.
https://github.com/mlr-org/mlr3book/blob/ac3f3b95eee0f4d1556734d83d05860596ad4c16/book/basics.qmd#L16
* [x] Remove, not useful
https://github.com/mlr-org/mlr3book/blob/ac3f3b95eee0f4d1556734d83d05860596ad4c16/book/basics.qmd#L25
* [x] This needs more consistency. If you're going to give a quick overview then make sure it's detailed enough to be understood, e.g. "unsupervised task to estimate the density" hasn't explained the task. Consider removing tasks in development.
https://github.com/mlr-org/mlr3book/blob/ac3f3b95eee0f4d1556734d83d05860596ad4c16/book/basics.qmd#L66
* [x] Too technical. This is covered in a later chapter
https://github.com/mlr-org/mlr3book/blob/ac3f3b95eee0f4d1556734d83d05860596ad4c16/book/basics.qmd#L104
* [x] Either in Chapter 1 or 2 (maybe 1) need to explain dictionaries
https://github.com/mlr-org/mlr3book/blob/ac3f3b95eee0f4d1556734d83d05860596ad4c16/book/basics.qmd#L134
* [x] This is putting down your own code design!
https://github.com/mlr-org/mlr3book/blob/ac3f3b95eee0f4d1556734d83d05860596ad4c16/book/basics.qmd#L137-L145
It's enough just to say
https://github.com/mlr-org/mlr3book/blob/ac3f3b95eee0f4d1556734d83d05860596ad4c16/book/basics.qmd#L143-L145
* [x] Not sure what this refers to
https://github.com/mlr-org/mlr3book/blob/ac3f3b95eee0f4d1556734d83d05860596ad4c16/book/basics.qmd#L147
* [x] Only refer to sugar. At the end of the chapter show the link to R6
https://github.com/mlr-org/mlr3book/blob/ac3f3b95eee0f4d1556734d83d05860596ad4c16/book/basics.qmd#L149-L151
* [x] Remove, not useful
https://github.com/mlr-org/mlr3book/blob/ac3f3b95eee0f4d1556734d83d05860596ad4c16/book/basics.qmd#L158
* [x] I found this very confusing. Need to better distinguish between objects and classes and maybe explain why you can't use `?`
https://github.com/mlr-org/mlr3book/blob/ac3f3b95eee0f4d1556734d83d05860596ad4c16/book/basics.qmd#L163-L170
* [ ] Not convinced this is the best way to present API as we don't really do this anywhere else. What's value-add here instead of sending people to online API? Better just to show a working example that actually explains what each method and field does or explain design decisions
https://github.com/mlr-org/mlr3book/blob/main/book/basics.qmd#L183
* [ ] Following from above, this warning isn't particularly useful. Instead you could show an example of subsetting where you'd actually use `row_ids` in practice and then explain the code
https://github.com/mlr-org/mlr3book/blob/ac3f3b95eee0f4d1556734d83d05860596ad4c16/book/basics.qmd#L218
* [x] "Training data" -> "Training task" use mlr language throughout. Separate regr/classif. Everywhere here use regr, later consider classif.
https://github.com/mlr-org/mlr3book/blob/ac3f3b95eee0f4d1556734d83d05860596ad4c16/book/basics.qmd#L411
* [x] Term "Predefined" a bit confusing. Need a discussion about where learners live: mlr3, mlr3learners,  mlr3extralearners
https://github.com/mlr-org/mlr3book/blob/ac3f3b95eee0f4d1556734d83d05860596ad4c16/book/basics.qmd#L420
* [ ] Same comment as Task API, unconvinced this is best use of space
https://github.com/mlr-org/mlr3book/blob/ac3f3b95eee0f4d1556734d83d05860596ad4c16/book/basics.qmd#L469
* [x] Too wordy, overly complex
https://github.com/mlr-org/mlr3book/blob/ac3f3b95eee0f4d1556734d83d05860596ad4c16/book/basics.qmd#L517-L520
* [x] More complex than needed, just show one worked example
https://github.com/mlr-org/mlr3book/blob/ac3f3b95eee0f4d1556734d83d05860596ad4c16/book/basics.qmd#L532
* [x] Overly complex, many people won't know what `fread` does, why are you defiing your own data anyway? Showcase the `partition` method and use `row_ids`!
https://github.com/mlr-org/mlr3book/blob/ac3f3b95eee0f4d1556734d83d05860596ad4c16/book/basics.qmd#L561
* [x] Do you really want to show `predict_newdata` before `predict`?
https://github.com/mlr-org/mlr3book/blob/ac3f3b95eee0f4d1556734d83d05860596ad4c16/book/basics.qmd#L573
* [x] As above, don't put down your software, this sounds like a design flaw. The easiest way should be to use your own design!! Also I think this might be old anyway because they print the same way.
https://github.com/mlr-org/mlr3book/blob/ac3f3b95eee0f4d1556734d83d05860596ad4c16/book/basics.qmd#L578
* [x] First put in separate classif section. Second, this confuses *a lot* of people. `prob` is clearly more informative and useful, why is it not the default? Take the opportunity to explain a common FAQ
https://github.com/mlr-org/mlr3book/blob/ac3f3b95eee0f4d1556734d83d05860596ad4c16/book/basics.qmd#L601
* [x] Inconsistent level of detail with last plotting section (I prefer this). Again show one plot then just list possibilities for different plot types
https://github.com/mlr-org/mlr3book/blob/ac3f3b95eee0f4d1556734d83d05860596ad4c16/book/basics.qmd#L732

## Structure

* [x] Strongly consider re-write as follows:
i. Tasks (focusing on regression tasks only)
ii. Learners (focusing on regression learners only)
iii. Evaluation (focusing on regression measures only)
iv. Classification
v. Putting it together: Train, Test, Evaluate
* [x] Remove here in favour of longer discussion in [Chapter 1](https://github.com/mlr-org/mlr3book/issues/437)
https://github.com/mlr-org/mlr3book/blob/ac3f3b95eee0f4d1556734d83d05860596ad4c16/book/basics.qmd#L126
* [x] Unconvinced TOC is needed. Limited use in published book and we don't have this in other chapters. Instead maybe give a 1-2 line summary of each section? 
https://github.com/mlr-org/mlr3book/blob/ac3f3b95eee0f4d1556734d83d05860596ad4c16/book/basics.qmd#L31
* [x] Move into separate classification section
https://github.com/mlr-org/mlr3book/blob/ac3f3b95eee0f4d1556734d83d05860596ad4c16/book/basics.qmd#L337
* [x] I don't think this needs so many examples and hopefully before this point we will have explained mlr3viz in intro, so maybe pick one plot and then just list `type` and what they plot
https://github.com/mlr-org/mlr3book/blob/ac3f3b95eee0f4d1556734d83d05860596ad4c16/book/basics.qmd#L356
* [x] Move into Measures section (and other associated content)
https://github.com/mlr-org/mlr3book/blob/ac3f3b95eee0f4d1556734d83d05860596ad4c16/book/basics.qmd#L680
* [x] Move into classification section
https://github.com/mlr-org/mlr3book/blob/ac3f3b95eee0f4d1556734d83d05860596ad4c16/book/basics.qmd#L633
https://github.com/mlr-org/mlr3book/blob/ac3f3b95eee0f4d1556734d83d05860596ad4c16/book/basics.qmd#L711

## Introduction (@pfistfl @sumny)

Organize the introduction around the following topics: 
- [ ] Motivation
- [ ] Basic Concepts 
    - [ ] What are the basic concepts we'll need to know? 
    - [ ] What are the `mlr3` objects we will interact with?
    - [ ] Can we visualize that nicely?
- [ ] Further Reading 
  - [ ] What topics within the book are relevant?
  - [ ] Point  to references and additional (outside) links

**Comment by larskotthoff** [comment]
Also describe thresholding for multiple classes.

**Comment by larskotthoff** [comment]
And add something at the end about mlr3extralearners and all the learners it supports.

**Comment by RaphaelS1** [comment]
Adding one final point about param_sets in measures from #236 

* [ ] We should document in the book how to set measure hyperparams and where to find them. I looked for quite a while how to set beta for classif.fbeta and then found the solution in the unit tests. I could also not find anything about measures having param_sets in the book ?

**Comment by RaphaelS1** [comment]
Copying https://github.com/mlr-org/mlr3book/issues/132 here.

* [ ] All plots should have some text explaining what we are seeing

**Comment by larskotthoff** [comment]
Consider renaming chapter "Fundamentals" as it's going quite a bit beyond the basics now.

**Comment by larskotthoff** [comment]
- basic intro to ML (here, not in chapter 1)
- roles and columns: move to end of chapter, explain all col roles, give some more examples (in particular for observation weights in weight column) and refs to chapters after that
- explain learner API for predict better, what do the arguments mean?
- add `as.data.table` for result of prediction

**Comment by RaphaelS1** [comment]
* [ ] Copied from MM: I think Fundamentals needs a new but very short section under "Learners" called something like "Parameters". We already have most of the content from this under "Training the learner" but I think here we need to expliclty say: 1) the param_set field returns a ParamSet class. 2) Each class contains ParamX which represent XXX. 3) We will look at creating parameter sets in Optimisation chapter.
